end_to_end/end_to_end_suite_test.go

```diff
@@ -2690,4 +2690,204 @@ LANGUAGE plpgsql NO SQL;`)
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb")
+
+			assertDataRestored(restoreConn, map[string]int{
+				"public.part_d": 20,
+			})
+
+			assertTablesNotRestored(restoreConn, []string{"part_c"})
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("ignore partitions specified in exclude-table during restore", func() {
+			testhelper.AssertQueryRuns(backupConn, `
+					CREATE TABLE part_c PARTITION OF t_part_1_prt_1 FOR VALUES IN ('c');
+					CREATE TABLE part_d PARTITION OF t_part_1_prt_1 FOR VALUES IN ('d');
+
+					INSERT INTO part_c SELECT a, 0, a, 'c' FROM generate_series(1, 10)a;
+					INSERT INTO part_d SELECT a, 0, a, 'd' FROM generate_series(1, 20)a;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir, "--leaf-partition-data")
+
+			//to prevent new partitions from being created during restoration, we need to exclude the root.
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb",
+				"--exclude-table", "public.t_part")
+			assertTablesNotRestored(restoreConn, []string{"part_c", "part_d"})
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("allows the user to exclude partitions that hinder restore if the default partition in the extension contains data", func() {
+			testhelper.AssertQueryRuns(backupConn, `
+					ALTER EXTENSION test_ext6 DROP TABLE d_part_1_prt_extra;
+					ALTER TABLE d_part SPLIT DEFAULT PARTITION START (10) END (20)
+						INTO (PARTITION new_part, default partition);
+					ALTER EXTENSION test_ext6 ADD TABLE d_part_1_prt_extra;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir, "--leaf-partition-data")
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb",
+				"--exclude-table", "public.d_part")
+
+			assertTablesNotRestored(restoreConn, []string{"d_part_1_prt_new_part"})
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("does not lose inheritance during backup if the parent table is in the extension", func() {
+			testhelper.AssertQueryRuns(backupConn, `
+				CREATE TABLE child (d int) INHERITS (parent);
+			`)
+
+			defer testhelper.AssertQueryRuns(backupConn, `DROP TABLE child;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir)
+			metadataFileContents := string(getMetdataFileContents(backupDir, timestamp, "metadata.sql"))
+			Expect(metadataFileContents).To(ContainSubstring("INHERITS (public.parent)"))
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb")
+
+			exists := make([]bool, 0)
+			err := restoreConn.Select(&exists,
+				"SELECT EXISTS(select * from pg_inherits where inhrelid = 'child'::regclass and inhparent = 'parent'::regclass);")
```

Comment:
The plan is simpler when `select from` is used instead of `select * from`. No target list is needed here. Let's write the key words in uppercase as in the code around.